ktor-server/ktor-server-core/jvm/src/io/ktor/config/ApplicationConfig.kt

```diff
@@ -39,18 +44,160 @@ interface ApplicationConfig {
  * Represents an application config value
  */
 @KtorExperimentalAPI
-interface ApplicationConfigValue {
+class HoconApplicationConfigValue(private val config: Config, private val path: String) {
     /**
-     * Get property string value
+     * If the value is empty
+     * @see [Config.isEmpty]
      */
-    fun getString(): String
+    fun isEmpty(): Boolean = config.isEmpty
 
     /**
-     * Get property list value
+     * Get property [Boolean] value
+     * @see [Config.getBoolean]
      */
-    fun getList(): List<String>
+    fun getBoolean(): Boolean = config.getBoolean(path)
+
+    /**
+     * Get property [Number] value
+     * @see [Config.getNumber]
+     */
+    fun getNumber(): Number = config.getNumber(path)
+
+    /**
+     * Get property [Int] value
+     * @see [Config.getInt]
+     */
+    fun getInt(): Int = config.getInt(path)
+
+    /**
+     * Get property [Long] value
+     * @see [Config.getLong]
+     */
+    fun getLong(): Long = config.getLong(path)
+
+    /**
+     * Get property [Double] value
+     * @see [Config.getDouble]
+     */
+    fun getDouble(): Double = config.getDouble(path)
+
+    /**
+     * Get property [String] value
+     * @see [Config.getString]
+     */
+    fun getString(): String = config.getString(path)
+
+    /**
+     * Get property enum value
+     * @param enumClass The type of enum to get the value as
+     * @see [Config.getEnum]
+     */
+    fun <T : Enum<T>> getEnum(enumClass: KClass<T>): T = config.getEnum(enumClass.java, path)
+
+    /**
+     * Get property as [Any]
+     * @see [Config.getAnyRef]
+     */
+    fun getAnyRef(): Any = config.getAnyRef(path)
+
+    /**
+     * Get property [ConfigMemorySize] value
+     * @see [Config.getMemorySize]
+     */
+    fun getMemorySize(): ConfigMemorySize = config.getMemorySize(path)
+
+    /**
+     * Get property [Duration] value
+     * @param timeUnit The [TimeUnit] to get the duration as
+     * @see [Config.getDuration]
+     */
+    fun getDuration(timeUnit: TimeUnit): Long = config.getDuration(path, timeUnit)
+
+    /**
+     * Get property [Duration] value
+     * @see [Config.getDuration]
+     */
+    fun getDuration(): Duration = config.getDuration(path)
```

Comment:
Even more: `Duration` requires Android API level 26. This is quite problematic, unfortunately.

Reply:
Wow, I'm glad I don't have to deal with mobile stuff when I read stuff like this! I guess this one could be removed.